bundles/org.eclipse.cdt.lsp/src/org/eclipse/cdt/lsp/internal/editor/FormatOnSave.java

```diff
@@ -15,16 +15,24 @@
 import org.eclipse.cdt.lsp.config.Configuration;
 import org.eclipse.cdt.lsp.editor.EditorOptions;
 import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.jface.preference.IPreferenceStore;
 import org.eclipse.jface.text.IDocument;
 import org.eclipse.jface.text.IRegion;
 import org.eclipse.lsp4e.LSPEclipseUtils;
+import org.eclipse.lsp4e.LanguageServerPlugin;
 import org.eclipse.lsp4e.format.IFormatRegionsProvider;
 import org.osgi.service.component.annotations.Component;
 import org.osgi.service.component.annotations.Reference;
 
 @Component(property = { "serverDefinitionId:String=org.eclipse.cdt.lsp.server" })
 public class FormatOnSave implements IFormatRegionsProvider {
 
+	public FormatOnSave() {
```

Comment:
Perhaps component constructor is not the best place to modify preferences, more common is to involve [preference initializer](https://help.eclipse.org/latest/index.jsp?topic=%2Forg.eclipse.platform.doc.isv%2Freference%2Fapi%2Forg%2Feclipse%2Fcore%2Fruntime%2Fpreferences%2FAbstractPreferenceInitializer.html) 

However, I'm not familiar enough with the way how LSP4E defines the default value of this preference.

Reply:
Your suggestion works, because the FormatOnSave class calls the `org.eclipse.cdt.lsp.config.Configuration` class. This class involves the PreferenceInitializer  class you mentioned. Please check my last commit.